File: backend/src/models/Medicine.py
from typing import Optional, List

from pydantic import BaseModel


class Medicine(BaseModel):
    medicine_id: Optional[str] = None
    image_src: Optional[str] = None
    english_name: str
    swahili_name: Optional[str] = None
    kikuyu_name: Optional[str] = None
    side_effects: List[str] = []
    treats: List[str] = []

    @staticmethod
    def from_dict(data: dict):
        return Medicine(
            medicine_id=data.get("medicine_id"),
            image_src=data.get("image_src"),
            english_name=data.get("english_name"),
            swahili_name=data.get("swahili_name"),
            kikuyu_name=data.get("kikuyu_name"),
            side_effects=data.get("side_effects"),
            treats=data.get("treats")
        )

    def to_dict(self):
        return {
            'medicine_id': self.medicine_id,
            'image_src': self.image_src,
            'english_name': self.english_name,
            'swahili_name': self.swahili_name,
            'kikuyu_name': self.kikuyu_name,
            'side_effects': self.side_effects,
            'treats': self.treats,
        }

File: web/src/models/Medicine.ts
// Medicine Class
import {AppLanguage} from "@/context/AppLanguageContext.ts";

export default class Medicine {
    medicine_id?: string;
    image_src?: string;
    english_name: string;
    swahili_name?: string;
    kikuyu_name?: string;
    side_effects: string[] = [];
    treats: string[] = [];

    constructor(
        english_name: string,
        side_effects: string[] = [],
        treats: string[] = [],
        medicine_id?: string,
        image_src?: string,
        swahili_name?: string,
        kikuyu_name?: string
    ) {
        this.medicine_id = medicine_id;
        this.image_src = image_src;
        this.english_name = english_name;
        this.swahili_name = swahili_name;
        this.kikuyu_name = kikuyu_name;
        this.side_effects = side_effects;
        this.treats = treats;
    }

    getTranslatedName(appLanguage: AppLanguage): string {
        switch (appLanguage) {
            case AppLanguage.English:
                return this.english_name
            case AppLanguage.Swahili:
                if (this.swahili_name !== undefined) {
                    return this.swahili_name
                } else {
                    return this.english_name
                }
            case AppLanguage.Kikuyu:
                if (this.kikuyu_name !== undefined) {
                    return this.kikuyu_name
                } else {
                    return this.english_name
                }
        }
    }
}
